openwisp_notifications/base/views.py

```diff
@@ -41,4 +52,93 @@ def test_func(self):
         return True
 
 
-notification_preference_page = NotificationPreferencePage.as_view()
+@method_decorator(csrf_exempt, name='dispatch')
+class UnsubscribeView(TemplateView):
+    template_name = 'openwisp_notifications/unsubscribe.html'
+
+    def get(self, request, *args, **kwargs):
+        encoded_token = request.GET.get('token')
+        if not encoded_token:
+            return render(request, self.template_name, {'valid': False})
+
+        user, valid = self._validate_token(encoded_token)
+        if not valid:
+            return render(request, self.template_name, {'valid': False})
+
+        is_subscribed = self.get_user_preference(user)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                'valid': True,
+                'user': user,
+                'is_subscribed': is_subscribed,
+            },
+        )
+
+    def post(self, request, *args, **kwargs):
+        encoded_token = request.GET.get('token')
+        if not encoded_token:
+            return JsonResponse(
+                {'success': False, 'message': 'No token provided'}, status=400
+            )
+
+        user, valid = self._validate_token(encoded_token)
+        if not valid:
+            return JsonResponse(
+                {'success': False, 'message': 'Invalid or expired token'}, status=400
+            )
+
+        subscribe = False
+        if request.body:
+            try:
+                data = json.loads(request.body)
+                subscribe = data.get('subscribe', False)
+            except json.JSONDecodeError:
+                return JsonResponse(
```

Comment:
I couldn't find any details on the required response for the POST request from email client. Did you return JSON just for the added JS logic in unsubscribe.js or is this a requirement?

Reply:
As discussed in our call, I have added selenium tests to test the API requests.